### PR TITLE
🟡 ⚫ Boosted dark mode prototype

### PR DESCRIPTION
**Pull request by julien-deramond**
This PR is a prototype to play with the first set of colors provided by the designers to define the dark mode for Boosted.
However, IMO it can be merged as-is since our current dark mode version is based on Bootstrap (their colors) and is broken in our environment, and hurt my eyes everyday 😄 

In light mode, we have components and their dark variants.
In dark mode, the light and dark variants are displayed the same way (with a black background)

Some rules for dark mode:
* Black background : `#000000`
* Default body text color: `#eeeeee`
* For the components, the text color must remain the same as for the dark variants, so in most cases: `#ffffff`

For the moment, all this behavior is not developed. Here is the list of things tried:
* Themes colors (primary, secondary, info, warning, danger) have the same values as in light mode
* Body text color (default text color in a page) is set to `$body-color-dark: #eee `
* Default background color is set to `$body-bg-dark: #000`
* Tertiary background color is set arbitrarily to `$body-tertiary-bg-dark: #414141`
  * Some fixes have been done in the documentation to avoid having white background, so I replaced `bg-white` by `bg-tertiary-bg` 

### [Live preview](https://deploy-preview-1858--boosted.netlify.app/)

### CODE REVIEW

File: scss/_variables-dark.scss [hunk 1]
```diff
@@ -8,44 +8,44 @@
...
+$info-border-subtle-dark:           $info !default; // Boosted mod: instead of `shade-color($info, 40%)`
+$warning-border-subtle-dark:        $warning !default; // Boosted mod: instead of `shade-color($warning, 40%)`
+$danger-border-subtle-dark:         $danger !default; // Boosted mod: instead of `shade-color($danger, 40%)`
+$light-border-subtle-dark:          $light !default; // Boosted mod: instead of `$gray-700`
+$dark-border-subtle-dark:           $dark !default; // Boosted mod: instead of `$gray-800`
 // scss-docs-end theme-border-subtle-dark-variables
 
-$body-color-dark:                   $gray-500 !default;
-$body-bg-dark:                      $gray-900 !default;
+$body-color-dark:                   $gray-300 !default; // Boosted mod: instead of `$gray-500`
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply scss/_variables-dark.scss:43]: Me too but this is the current value suggested by the design.

**Review by louismaximepiton — APPROVED** [review]
Tried to tackle the things that were not hard to change inside this PR, it's up to you if find it fine or not !

File: scss/_variables-dark.scss [hunk 2]
```diff
@@ -46,12 +46,12 @@ $body-secondary-color-dark:         rgba($body-color-dark, .75) !default;
...
-$body-emphasis-color-dark:          $gray-100 !default;
+$body-emphasis-color-dark:          $white !default; // Boosted mod: instead of `$gray-100`
 $border-color-dark:                 $gray-700 !default;
-$border-color-translucent-dark:     rgba($white, .15) !default;
+$border-color-translucent-dark:     $gray-700 !default; // Boosted mod instead of `rgba($white, .15)`
 $headings-color-dark:               null !default;
-$link-color-dark:                   tint-color($primary, 40%) !default;
-$link-hover-color-dark:             shift-color($link-color-dark, -$link-shade-percentage) !default;
+$link-color-dark:                   $white !default; // Boosted mod: instead of `tint-color($primary, 40%)`
+$link-hover-color-dark:             $primary !default; // Boosted mod: instead of `shift-color($link-color-dark, -$link-shade-percentage)`
```
Comment (julien-deramond) [inline scss/_variables-dark.scss:54]: Not supposed to be `$accessible-orange` here?

File: scss/_variables.scss [hunk 3]
```diff
@@ -646,13 +646,13 @@ $outline-offset:              $outline-width !default; // Deprecated in v5.2.3
 $focus-visible-zindex:                5 !default; // Boosted mod
 
 $focus-visible-inner-width:           2px !default; // Boosted mod
-$focus-visible-inner-color:           $white !default; // Boosted mod
-$focus-visible-inner-color-inverted:  $black !default; // Boosted mod
+$focus-visible-inner-color:           var(--#{$prefix}body-bg) !default; // Boosted mod
```
Comment (julien-deramond) [inline scss/_variables.scss:649]: Not a big fan of using a "background" color to define a text color. If no other custom vars are possible, we can let it as is for now. (several occurrences in this PR)

File: site/assets/scss/_boosted.scss [hunk 4]
```diff
@@ -60,7 +60,7 @@ body {
 // Design guidelines
 
 .ods-guidelines {
-  background-color: $gray-300;
+  background-color: var(--bs-tertiary-bg);
```
Comment (julien-deramond) [inline site/assets/scss/_boosted.scss:63]: This is not the right color in light mode as defined by the design guidelines

**Review by julien-deramond — COMMENTED** [review]
@louismaximepiton Few comments to tackle together. But thanks for these modifications :)

File: scss/_variables.scss (hunk 3 above)
Comment (julien-deramond) [reply scss/_variables.scss:649]: Modification to do later, tracked in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1776

File: site/assets/scss/_boosted.scss (hunk 4 above)
Comment (julien-deramond) [reply site/assets/scss/_boosted.scss]: Fixed by using `var(--bs-secondary-bg)`

**Pull request merged**